pkg/engine/task/task.go

```diff
@@ -53,7 +54,7 @@ func Build(task *v1alpha1.Task) (Tasker, error) {
 	case DummyTaskKind:
 		return newDummy(task), nil
 	default:
-		return nil, fmt.Errorf("%wunknown task kind %s", ErrFatalExecution, task.Kind)
+		return nil, fmt.Errorf("%wunknown task kind %s", engine.ErrFatalExecution, task.Kind)
```

Comment:
so this one is fatal but not an execution error? 🤔 is that intentional? I think we should publish event here as well

Reply:
We do that one level above, where this method is called. But we should make this error fatal there so 👍